Move libguestfs apis to libguestfs_api.py

zcyes opened the pull request.

yumingfei commented:
Hey, @zcyes:
It's really a vast patch. You can try to finish this job in several patches to make the steps more clear.
BTW, I don't think the module utils_test is a good place to put these APIs in.
Because these APIs are useful and maybe we can use it in some other libvirt modules.
While utils_test is mainly used for test modules in tp-libvirt.
What do you think?

zcyes commented:
Hey, @yumingfei :
Tank you for your suggestion.
1. Vast patch
Indeed, It's a vast patch, contains so many "deletes" and "adds". However, There is just one step in the job actually, The steps are: Just make a new file named "libguestfs_api.py", and move the apis into the new file. Hence, most of the "changes" are moving apis to new file. So "finish this job in serveral patches" is not very necessary. :)
2. APIs
It's a good advice. we thinks three solutions.
a) move the apis to a new file in the folder where utils_libguestfs.py locates. And move the apis into the new file.
b) move the apis to a new class defined in utils_libguestfs.py. and move apis to the new class. However, this is not a good solution, cause we don't reduce the scale of utils_libguestfs.py.
c) make a new folder in the folder where utils_libguestfs.py locates, then make a new file in the new folder, and move the apis to the new file. This is not a good solution too, cause we make a new folder just for one file.
What do you think? Thanks

yumingfei commented:
How about making a new fold named utils_libguestfs in virttest:
```
virt-test---xxx
             |----utils_libguestfs----__init__.py
                                            |--guestfish_api.py
                                            |--libguestfs_tools.py
                                            \--xxx
```
Then we can put all libguestfs about functions into it and it would be much easy to extend.

zcyes commented:
This pull request is made for libguestfs apis.
We choose @yumingfei suggestion, and the folder view is:
virt-test|---xxx
            |---virttest|---xxx
                           |---utils_libguestfs|---__init__.py
                                                       |---utils_libguestfs.py 
                                                       |---libguestfs_api.py 
the steps are:
1. make a new folder virttest/utils_libguestfs
2. move the apis from virttest/utils_libguestfs.py into virttest/utils_libguestfs/libguestfs_api.py
3. move virttest/utils_libguestfs.py ---> virttest/utils_libguestfs/utils_libguestfs.py

What do you think about? thank you.

yumingfei commented:
Seems that your PR need to be updated for meger.
BTW, just FYI:
One trick -- you can use __init__.py to put some importing steps just like libvirt_xml module.
Then we do not need to fix test module for those imports.

Commit:
Move libguestfs apis to libguestfs_api.py

Commit:
move libguestfs apis to new folder
1. make a new folder virttest/utils_libguestfs
2. move libguestfs about files(utils_libguestfs.py, libguestfs_api.py) into the virttest/utils_libguestfs

Commit:
Move some import steps into init.py

Commit:
Modify the import format

Commit:
Modify virttest/utils_libguestfs_unittest.py for new apis location

Commit:
Move utils_libguestfs_unittest.py into virttest/utils_libguestfs

shi2wei3 commented:
Run part of test cases of libguestfs and test result is PASS, code looks good to me.

Only two things needs @zcyes 's attention
1. The first patch is not comply with integrity constraints, if somebody only apply the first patch, it will cause all test cases failed. Please pay attention next time.
2. Move utils_libguestfs_unittest.py to  utils_libguestfs

@yumingfei This patch involve common file adjustments, we also need your ack :-)

Changes summary
.
.......
└── virttest
.......
    ├── utils_libguestfs
    │   ├── __init__.py (new added)
    │   ├── libguestfs_api.py (new added, contains APIs)
    │   ├── utils_libguestfs.py (move to here, APIs moved to libguestfs_api.py)
    │   └── utils_libguestfs_unittest.py (move to here)
.......
    ├── utils_libguestfs.py (moved to utils_libguestfs)
    ├── utils_libguestfs_unittest.py (moved to utils_libguestfs)


class inheritance relationship
Before:
GuestfishPersistent -> GuestfishTools

After:
GuestfishPersistent -> GuestfishAPIs -> GuestfishTools

shi2wei3 commented:
https://github.com/autotest/tp-libvirt/pull/446 is pending on this pull request

yumingfei commented on virttest/utils_libguestfs/__init__.py:
```diff
@@ -0,0 +1,9 @@
+from virttest.utils_libguestfs.libguestfs_api import GuestfishAPIs
+
+from virttest.utils_libguestfs import utils_libguestfs
```
It's a little confused by importing xxx.utils_libguestfs from utils_libguestfs..

yumingfei commented:
Hey, @zcyes @shi2wei3 
.......
└── virttest
.......
├── utils_libguestfs
│   ├── init.py (new added)
│   ├── libguestfs_api.py (new added, contains APIs)
Maybe guestfish_api is more suitable?
│   ├── utils_libguestfs.py (move to here, APIs moved to libguestfs_api.py)
│   └── utils_libguestfs_unittest.py (move to here)
__How about base_libguestfs instead of utils_libguestfs?__
And I think we can move virt_xxx_cmd functions out from it just like libguestfs_api?
.......
├── utils_libguestfs.py (moved to utils_libguestfs)
├── utils_libguestfs_unittest.py (moved to utils_libguestfs)

shi2wei3 commented:
How about base_libguestfs instead of utils_libguestfs?
A: make sense, will do it.

And I think we can move virt_xxx_cmd functions out from it just like libguestfs_api?
A: good idea, we will do this later and open a new pull request to handle virt-tools stripping

shi2wei3 commented:
Hi @yumingfei , i will merge this pull request once you ack it.
Since new APIs development will use these files, let's handle this ASAP :-)

yumingfei commented:
Others look fine to me.
Rome was not built in one day. :D

Commit:
Rename utils_libguestfs.py to base_libguestfs.py

Commit:
Rename libguestfs_api.py to guestfish_api.py.py

shi2wei3 commented:
@yumingfei Not means to push you :-)
zcyes will leave for more than one month, so i want to close his pull requests.
BTW, i'll take paid time off next week.
Do it when i come back is also OK.

yumingfei commented:
Just leave with it is open, maybe others have idea on it. :)

Commit:
Add selinux apis

Commit:
Add mke2fs apis

Commit:
Modify guestmount

lmr commented:
Hello @zcyes, if you could be so kind, please rebase it and re-open it against

https://github.com/avocado-framework/avocado-vt

Thanks!
